### PR TITLE
Fix: Stop using UIScreen.main

### LINKED ISSUE #2741: Initializing Sentry in a SwiftUI app results in accent color not being respected
### Platform

iOS

### Installed

Swift Package Manager

### Version

8.2.0

### Steps to Reproduce

1. Set a custom accent color (e.g., red) by modifying the `AccentColor` in `Assets.xcassets`
2. Create a view that uses the above accent color in some way (e.g., the `foregroundColor` of a `Text` view)
3. Add the following line to the `init` of the `App`-conforming struct identified by `@main`
`init() { SentrySDK.start { _ in } }`
5. Run the app

OR

Download and run this [sample project](https://github.com/caseybutatwork/SentrySwiftUIAccentColor)

### Expected Result

The text is red (the custom accent color)

### Actual Result

The text is blue (the default accent color)

### Are you willing to submit a PR?

Unfortunately, I do not have the time

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

UIScreen.main is deprecated, and using it is causing problems with the accent color for SwiftUI.

## :bulb: Motivation and Context

Closes #2741 

## :green_heart: How did you test it?

Samples

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/Sentry/SentryCrashIntegration.m [hunk 1]
```diff
@@ -275,8 +276,15 @@ + (void)enrichScope:(SentryScope *)scope crashWrapper:(SentryCrashWrapper *)cras
 #if SENTRY_HAS_UIDEVICE && !defined(TESTCI)
     // Acessessing UIScreen.mainScreen fails when using SentryTestObserver.
     // It's a bug with the iOS 15 and 16 simulator, it runs fine with iOS 14.
```
Comment (brustolin) [inline Sources/Sentry/SentryCrashIntegration.m:278]:
```suggestion
```

File: Sources/Sentry/SentryCrashIntegration.m [hunk 2]
```diff
@@ -273,10 +274,15 @@ + (void)enrichScope:(SentryScope *)scope crashWrapper:(SentryCrashWrapper *)cras
     [deviceData setValue:locale forKey:LOCALE_KEY];
 
 #if SENTRY_HAS_UIDEVICE && !defined(TESTCI)
-    // Acessessing UIScreen.mainScreen fails when using SentryTestObserver.
-    // It's a bug with the iOS 15 and 16 simulator, it runs fine with iOS 14.
```
Comment (armcknight) [inline Sources/Sentry/SentryCrashIntegration.m:277]: Does the fact that this is no longer the case mean that we can now remove the `!defined(TESTCI)` directive and run this logic in tests, if only for UI tests?

**Review by armcknight — APPROVED** [review]
LGTM

File: Sources/Sentry/SentryCrashIntegration.m (hunk 2 above)
Comment (brustolin) [reply Sources/Sentry/SentryCrashIntegration.m:277]: True, I guess we could.

**Comment by philipphofmann** [comment]
Funny bug. Thanks for fixing it, @brustolin 👏 .

**Pull request merged**